### DISCUSSION TITLE
Feature Request: Adding a Simple API to Get the Current Provider Name in the Active Chat Buffer.

### DISCUSSION THREAD

**Discussion by bassamsdata** [olimorris/codecompanion.nvim#116] [Ideas]
Thank you for this amazing plugin.

I'm trying to display the name of the current provider in the statusline so I can easily see which one I'm interacting with.

**Why:** I often switch between two LLMs within the same chat, and it would be really helpful to know which one will output the message without needing to press any keys. (it could be anything from a function could be accessed to a buffer variable like `vim.b.current_provider`.

Thank you!

**Comment by olimorris** [comment]
This is a great idea. I'm thinking a global variable `vim.g.codecompanion_current_adapter` may do the trick? What do you think?

**Reply by olimorris** [reply]
Also, how are you switching LLMs, is that with the `ga` keymap? I'm wondering if we fire an event that signifies an adapter has been changed/selected

**Reply by bassamsdata** [reply]
> I'm thinking a global variable vim.g.codecompanion_current_adapter may do the trick?

Thank you for the prompt response. I initially suggested using a buffer-local variable because people might have different chats with various LLMs. However, after some thought, I agree that a global variable like `vim.g.codecompanion_current_adapter` would be a better approach based on the currently focused chat buffer.


> how are you switching LLMs, is that with the ga keymap?

yes, I switch using `ga`. It's the only way I know to switch LLMs on the fly tbh.

> I'm wondering if we fire an event that signifies an adapter has been changed/selected

This would be great and probably necessary for the global API so that the statusline updates immediately when the event occurs.

Thank you

**Reply by olimorris** [reply]
Just pushed a commit (8561dc7ebe6f02abd4997d1d3973d0c3cea1424f) which adds the `CodeCompanionChatAdapter` event. When the chat buffer is created, the adapter is changed or the chat buffer is closed, the event will fire with a payload consisting of the buffer number and the adapter itself. As per the README, you can then hook into that for your statusline. Thought I'd [share](https://github.com/olimorris/dotfiles/blob/e0a686cf8715d98da462db92d454cced3e330ba7/.config/nvim/lua/plugins/heirline/statusline.lua#L479) what I do in mine.

**Reply by bassamsdata** [reply]
Thank you, it worked like a charm. Here's my implementation in case anyone needs it or wants to improve it.

```lua
local function get_llm_name()
vim.api.nvim_create_autocmd("User", {
  pattern = "CodeCompanionChatAdapter",
  callback = function(args)
    if args.data.adapter == nil or vim.tbl_isempty(args.data) then
      return
    end
    vim.g.llm_name = args.data.adapter.name
  end,
})
  if vim.g.llm_name == nil then
    return ""
  end
  if vim.bo.filetype == "codecompanion" then
  return string.format("%%#StatusLineLSP#[%s]", vim.g.llm_name)
  else
  return ""
  end
end

```
